vconfig/viper.go

```diff
@@ -66,3 +82,31 @@ func setWebsocketDefaults() {
 	viper.SetDefault(EnvWebsocketHistoryMax, 300)
 	viper.SetDefault(EnvWebsocketHistoryTtl, 10)
 }
+func setP2PDefaults() {
+	viper.SetDefault(EnvP2PLogLevel, p2pconfig.DefaultLogLevel)
+	viper.SetDefault(EnvP2PMaxPeers, p2pconfig.DefaultMaxPeers)
+	viper.SetDefault(EnvP2PMaxPeersPerIP, p2pconfig.DefaultMaxPeersPerIP)
+	viper.SetDefault(EnvP2PMinSyncPeerNetworkSpeed, p2pconfig.DefaultMinSyncPeerNetworkSpeed)
+	viper.SetDefault(EnvP2PBanDuration, p2pconfig.DefaultBanDuration)
+	viper.SetDefault(EnvP2PLogDir, p2pconfig.DefaultLogDir)
+	viper.SetDefault(EnvP2PExcessiveBlockSize, p2pconfig.DefaultExcessiveBlockSize)
+	viper.SetDefault(EnvP2PTrickleInterval, p2pconfig.DefaultTrickleInterval)
+	viper.SetDefault(EnvP2PBlocksForForkConfirmation, p2pconfig.DefaultBlocksToConfirmFork)
+}
+
+// ParseP2PConfig init p2p viper config based from specific config file.
+func ParseP2PConfig(cliCfg *p2pconfig.Config) *p2pconfig.Config {
+	viper.SetConfigFile(cliCfg.ConfigFile)
+	if err := viper.ReadInConfig(); err != nil {
+		err = fmt.Errorf("config can't read %s", err.Error())
+		log.Fatal(err)
+	}
+
+	p2pCfg := new(p2pconfig.Config)
+
+	if err := viper.Unmarshal(&p2pCfg); err != nil {
```

Comment:
Maybe we could actually unmarshal whole configuration with nested groups, and stop accessing values by keys?

Reply:
Done. Now we unmarshall all configs, not only p2p